src/nodes/manager/service.go

```diff
@@ -115,19 +120,19 @@ func (m *BaseManager) Node(_ context.Context, name string, userInfo *authtypes.U
 	m.mux.RLock()
 	defer m.mux.RUnlock()
 	if nodeBundle, ok := m.nodes[name]; ok {
-		if err := userInfo.CheckAccess(nodeBundle.manifest); err != nil {
-			errMsg := fmt.Sprintf("cannot access node '%s'", name)
-			m.logger.WithError(err).Warn(errMsg)
-			return nil, errors.FromError(err).SetMessage(errMsg)
+		permissions := m.authManager.UserPermissions(userInfo)
+		resolver := authorizator.New(permissions, userInfo.Tenant, m.logger)
+
+		err := resolver.CheckAccess(nodeBundle.manifest.AllowedTenants)
+		if err != nil {
+			return nil, err
 		}
-		return nodeBundle.node, nodeBundle.err
-	}
 
-	// This piece of code is here to make sure it is possible to retrieve a default node
-	for _, nodeBundle := range m.nodes {
-		if err := userInfo.CheckAccess(nodeBundle.manifest); err != nil {
-			continue
+		err = resolver.CheckPermission(&authtypes.Operation{Action: authtypes.ActionProxy, Resource: authtypes.ResourceNode})
```

Comment:
[IMPORTANT] You cannot check `proxy:node` permission here as you don't know whether or not the node will be use to proxy a call when you call this method Node(...). Instead the should be check at the Interceptor component where the proxy action really happens

Reply:
Since current implementation is pending of a refactor to include some intermedidate components to allow us to create fine grane permissions checks we decided over keeping this appoach as good for now.